### PR TITLE
fix(forge): fix `forge clone` when there is nested `src`

### LINKED ISSUE #7736: forge clone compiling issues (src mapping seems to break imports)
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (844caa8 2024-04-20T00:17:56.919599675Z)

### What command(s) is the bug in?

forge clone

### Operating System

Linux

### Describe the bug

I execute: 
```
forge clone  0x9d27527Ada2CF29fBDAB2973cfa243845a08Bd3F
```

I get

```
Compiler run failed:
Error (6275): Source "src/src/@openzeppelin/contracts/utils/introspection/IERC165.sol" not found: File not found.
 --> src/@openzeppelin/contracts/interfaces/IERC165.sol:5:1:
  |
5 | import "../utils/introspection/IERC165.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "src/src/@openzeppelin/contracts/utils/introspection/IERC165.sol" not found: File not found.
 --> src/@openzeppelin/contracts/token/ERC721/IERC721.sol:5:1:
  |
5 | import "../../utils/introspection/IERC165.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
...
```

even though the import path is fine

### PR DESCRIPTION
**Pull request by ZhangZhuoSJTU**
Closes #7736

### CODE REVIEW

**Comment by ZhangZhuoSJTU** [comment]
More local tests may be needed before merging it.

**Comment by ZhangZhuoSJTU** [comment]
I think the referenced bug has been properly fixed. @mattsse

File: crates/forge/bin/cmd/clone.rs [hunk 1]
```diff
@@ -407,74 +414,148 @@ fn update_config_by_metadata(
...
+                folder_name == "lib" ||
+                folder_name == "contracts" ||
+                folder_name == "hardhat" ||
+                folder_name == "forge-std" ||
+                folder_name.to_string_lossy().starts_with('@')
+        });
+
+    // ensure `src` and `lib` directories exist
+    eyre::ensure!(std::fs::metadata(root.join(src_dir)).is_ok(), "source directory must exists");
+    eyre::ensure!(std::fs::metadata(root.join(lib_dir)).is_ok(), "source directory must exists");
```
Comment (DaniPopes) [inline crates/forge/bin/cmd/clone.rs:460]: Any reason not to use `Path::exists` instead of `metadata.is_ok()`? The former is more readable.

Comment (ZhangZhuoSJTU) [reply crates/forge/bin/cmd/clone.rs:460]: Oh, yes, `Path::exists` is much better. I just copied-and-pasted my previous code and forgot this. Will fix.

**Review by DaniPopes — APPROVED** [review]
lgtm, pending @mattsse

**Review by mattsse — APPROVED** [review]

**Pull request merged**